CODEOWNERS.md

```diff
@@ -0,0 +1,11 @@
+# Plutus Codeowners
```

Comment:
I just realised something: this will block merging unless it gets reviewed by the specific code owner, right? Since you're the only code owner for the nix code, it might make sense to add someone else as well in case you're unavailable.

Reply:
Interesting questions, I'm not sure. 
I also don't know how CODEOWNER rules will play with existing PR & Merge rules.
We can just try it like this and see what happens?

Reply:
Does the OSC policy require there be a code owner for everything? I doubt adding `plutus-core-maintainers` to every PR is a good idea.